src/GraphQL/Execution/SubscriptionExecutionStrategy.cs

```diff
@@ -0,0 +1,296 @@
+            {
+                throw;
+            }
+            catch (ExecutionError error)
+            {
+                error.Path = node.ResponsePath;
+                error.AddLocation(node.Field, context.Document);
+                context.Errors.Add(error);
+                return null;
+            }
+            catch (Exception exception)
+            {
+                context.Errors.Add(await HandleExceptionInternalAsync(context, node, exception,
+                    $"Could not subscribe to field '{node.Field.Name}'.").ConfigureAwait(false));
+                return null;
+            }
+
+            // cannot throw an exception here
+            return subscription
+                .SelectCatchAsync(
+                    async (value, token) =>
+                    {
+                        // duplicate context to prevent multiple event streams from sharing the same context,
+                        // clear errors/metrics/extensions, and free array pool leased arrays
+                        using var childContext = new ExecutionContext(context)
+                        {
+                            Errors = new ExecutionErrors(),
+                            OutputExtensions = new Dictionary<string, object?>(),
+                            Metrics = Instrumentation.Metrics.None,
+                            CancellationToken = token,
+                        };
+
+                        return await ProcessDataAsync(childContext, node, value).ConfigureAwait(false);
+                    },
+                    async (exception, token) =>
+                    {
+                        using var childContext = new ExecutionContext(context)
+                        {
+                            Errors = new ExecutionErrors(),
+                            OutputExtensions = new Dictionary<string, object?>(),
+                            Metrics = Instrumentation.Metrics.None,
+                            CancellationToken = token,
+                        };
+
+                        return await ProcessErrorAsync(context, node, exception).ConfigureAwait(false);
+                    });
+        }
+
+        /// <summary>
+        /// Processes data from the event source via <see cref="IObserver{T}.OnNext(T)"/> and
+        /// returns an <see cref="ExecutionResult"/>.
+        /// <br/><br/>
+        /// Override this method to mutate <see cref="ExecutionContext"/> as necessary, such
+        /// as changing the <see cref="ExecutionContext.RequestServices"/> property to a scoped instance.
+        /// </summary>
+        protected virtual async Task<ExecutionResult> ProcessDataAsync(ExecutionContext context, ExecutionNode node, object? value)
+        {
+            var result = new ExecutionResult(context);
+
+            try
+            {
+                var executionNode = BuildSubscriptionExecutionNode(node.Parent!, node.GraphType!, node.Field, node.FieldDefinition, node.IndexInParentNode, value!);
```

Comment:
Why do you create execution node before listeners loop and not after?

Reply:
I'll look.  Probably could be changed.

Reply:
Changed

Reply:
Reverted; see below.

Reply:
>  // "clone" the node and set the source
                // overwrite the 'node' variable here so it is picked up by exception handling code below
                // and will contain the source from this data event

This?

Reply:
I was referring to those changes, as described below, supporting examining `node.Source` from the error handler.  Did you have a question?

Reply:
I already checked ~40 emails so I understood :)